contributors/design-proposals/storage/csi-migration.md

```diff
@@ -661,6 +695,93 @@ with minimal disruption to users would be:
 And at this point users doing their own deployment and not installing the GCE PD
 CSI driver encounter an error.
 
+## Disabling in-tree plugin code
+
+Before we can stop compiling and prepare to remove the code associated with migrated
+in-tree plugins, we need to make sure all persistent volume operations involving
+in-tree plugins continue to function in a backward compatible way when the
+in-tree plugin code paths are disabled. When CSIMigrationInTreeOff feature flag
+is enabled, we will disable in-tree plugin code by skipping probing of migrated
+in-tree plugins in Kubelet's ProbeVolumePlugins or Kube Controller Manager's
+ProbeAttachableVolumePlugins, ProbeExpandableVolumePlugins and
+ProbeControllerVolumePlugins. All functions in kubernetes code base that depend
+on probed plugins need to be audited and modified to handle nil plugins returned
+as detailed below.
+
+### Enhancements in ProbeVolumePlugins
+
+A new file pkg/volume/util/csimigration/plugins.go will have the following functions:
+```
+func ProbePluginsWithCSIMigrationDisabled(plugins []volume.VolumePlugin) []volume.VolumePlugin
+func ProbeAttachablePluginsWithCSIMigrationDisabled(plugins []volume.VolumePlugin) []volume.VolumePlugin
```

Comment:
Note: the main reason we need a separate ProbeAttachablePluginsWithCSIMigrationDisabled is because OpenShift Cinder does not implement the Attachable interface.

Reply:
@ddebroy which repo is `OpenShift Cinder`?

Reply:
do existing probe functions special case for `Cinder`? Want to avoid this if possible